### PR TITLE
Contributors guide

### PR DESCRIPTION
**Pull request by sengleson-r7**
### Assessment

* Updated readme for better navigation of the repo
* Added a Contributor's Guide to provide specific instructions for contributions
* Updated CONTRIBUTING.md to more accurately reflect best practices and available contribution methods
* Updated PULL_REQUEST_TEMPLATE.md to more accurately reflect requirements for PRs
* Added zip file to help workflow contributors get started
* Added screenshot for workflow validator

### CODE REVIEW

File: Contributors_Guide.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+# Workflow Contributor's Guide
+
+This page includes all the information needed to contribute an InsightConnect workflow to the Rapid7 Extension Library. For information on workflow building, please refer to the InsightConnect Help Documentation.
+
+_Note that contributing a workflow from InsightConnect does not share any of the credentials, API keys, passwords, or other user data utilized by your Connections!_ The .icon file exported from InsightConnect only contains the workflow logic itself -- the plugins, actions, decisions, filters, etc.
+
+There are three  to contributing a workflow:
```
Comment (jschipp-r7) [inline Contributors_Guide.md:7]:
```suggestion
There are three steps to contributing a workflow:
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,43 +1,37 @@
...
-- [ ] Create a directory called `screenshots` in your workflow's/snippet's directory
-- [ ] Includes PNG screenshots of the workflow in canvas, job, artifact in the `screenshots` directory e.g.  `workflows/Basic_Domain_Enrichment_Report/screenshots/`
-- [ ] PNG screenshot file names should conform to style e.g. `workflow1.png`, `artifact1.png`, `job.png`
-- [ ] Workflow verified as working through WUI
-- [ ] Credentials are not exported and stored in git
-- [ ] `workflow.spec.yaml` file with metadata accompanies workflow
-- [ ] Workflow documentation is provided in `help.md`
-- [ ] Directory structure is followed, see https://github.com/rapid7/insightconnect-workflows/blob/master/README.md
+- [ ] Workflow verified as working in a test
+- [ ] Workflow bundle includes `Name_Of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
```
Comment (jschipp-r7) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,43 +1,37 @@
...
-- [ ] Workflow documentation is provided in `help.md`
-- [ ] Directory structure is followed, see https://github.com/rapid7/insightconnect-workflows/blob/master/README.md
+- [ ] Workflow verified as working in a test
+- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
+- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
+- [ ] Workflow includes a detailed Description
+- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
+- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
+- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
+- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_Of_Workflow/`
```
Comment (jschipp-r7) [inline .github/PULL_REQUEST_TEMPLATE.md:13]:
```suggestion
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
```

File: Contributors_Guide.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+* A `screenshots` folder
+* A screenshot of the workflow in the Builder view in the screenshots folder
+* A screenshot of a successful job in the screenshots folder
+* A screenshot of the Final Report artifact after a successful execution in the screenshots folder
+
+Workflow bundles are organized according to the following directory structure:
+```
+└── workflows
+    └── Name_Of_Workflow
+        ├── Name_Of_Workflow.icon
```
Comment (jschipp-r7) [inline Contributors_Guide.md:48]:
```suggestion
        ├── Name_of_Workflow.icon
```

File: Contributors_Guide.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+![Workflow Validator Tool](./imgs/workflow_validator.png)
+
+## Step 3: Submit a Pull Request
+
+Once your workflow bundle passes the validation test, you're all set to submit a pull request! Please remember our *Dos* and *Don'ts*:
+
+* Do create a topic branch to work on. This helps ensure users are aware of commits on the branch being considered for merge, allows for a location for more commits to be offered without mingling with other contributor changes, and allows contributors to make progress while a PR is still being reviewed.
+* *Do* license your code as MIT.
+* *Do* follow the 50/72 rule for Git commit messages.
```
[the remark this answers is not in the document]
Comment (sengleson-r7) [reply Contributors_Guide.md:137]: adding a link to
https://stackoverflow.com/questions/2290016/git-commit-messages-50-72-formatting

**Pull request merged**